src/main/java/de/tum/in/www1/artemis/repository/metrics/ExerciseMetricsRepository.java

```diff
@@ -89,47 +89,45 @@ CASE WHEN TYPE(e) = ProgrammingExercise THEN TREAT(e AS ProgrammingExercise).all
      *
-     * @param exerciseIds the ids of the exercises
-     * @param userId      the id of the user
-     * @return the latest submission dates for the user in the exercises
+     * @param exerciseIds the ids of the exercises for which to fetch the latest submission dates
+     * @param userId      the id of the user whose submission dates are being fetched
+     * @return a set of ResourceTimestampDTO objects containing the exercise id and the latest submission date for the user
      */
     @Query("""
-            SELECT new de.tum.in.www1.artemis.web.rest.dto.metrics.ResourceTimestampDTO(e.id, s.submissionDate)
+            SELECT new de.tum.in.www1.artemis.web.rest.dto.metrics.ResourceTimestampDTO(e.id, MAX(s.submissionDate))
             FROM Submission s
                 LEFT JOIN StudentParticipation p ON s.participation.id = p.id
                 LEFT JOIN p.exercise e
                 LEFT JOIN p.team t
                 LEFT JOIN t.students u
             WHERE e.id IN :exerciseIds
-                AND s.submissionDate = (
-                    SELECT MAX(s2.submissionDate)
-                    FROM Submission s2
-                    WHERE s2.participation.id = s.participation.id
-                        AND s2.submitted = TRUE
-                )
+                AND s.submitted = TRUE
                 AND (p.student.id = :userId OR u.id = :userId)
+            GROUP BY p.id
             """)
     Set<ResourceTimestampDTO> findLatestSubmissionDatesForUser(@Param("exerciseIds") Set<Long> exerciseIds, @Param("userId") long userId);
 
     /**
      * Get the latest submission dates for a set of exercises.
+     * <p>
+     * This query fetches all latest submission dates for the specified set of exercises.
+     * It considers all participations related to the exercises.
      *
-     * @param exerciseIds the ids of the exercises
-     * @return the latest submission dates for the exercises
+     * @param exerciseIds the ids of the exercises for which to fetch the latest submission dates
+     * @return a set of ResourceTimestampDTO objects containing the exercise id and the latest submission date for each exercise
      */
     @Query("""
-            SELECT new de.tum.in.www1.artemis.web.rest.dto.metrics.ResourceTimestampDTO(e.id, s.submissionDate)
+            SELECT new de.tum.in.www1.artemis.web.rest.dto.metrics.ResourceTimestampDTO(e.id, MAX(s.submissionDate))
             FROM Submission s
                 LEFT JOIN StudentParticipation p ON s.participation.id = p.id
                 LEFT JOIN p.exercise e
             WHERE e.id IN :exerciseIds
-                AND s.submissionDate = (
-                    SELECT MAX(s2.submissionDate)
-                    FROM Submission s2
-                    WHERE s2.participation.id = s.participation.id
-                        AND s2.submitted = TRUE
-                )
+                AND s.submitted = TRUE
+            GROUP BY p.id
```

Comment:
ERROR: column \"e1_0.id\" must appear in the GROUP BY clause or be used in an aggregate function